### ISSUE TITLE
Checksum bug newpipe

### ISSUE THREAD

**Issue by Brimont** [TeamNewPipe/NewPipe#2077]
MD5 bug
## Exception
* __User Action:__ ui error
* __Request:__ App crash, UI failure
* __Content Language:__ IT
* __Service:__ none
* __Version:__ 0.15.1

<details><summary><b>Crash log</b></summary><p>

```
java.lang.ClassCastException: org.schabi.newpipe.App cannot be cast to android.app.Activity
	at us.shandian.giga.ui.adapter.MissionAdapter$ChecksumTask.<init>(MissionAdapter.java:721)
	at us.shandian.giga.ui.adapter.MissionAdapter.handlePopupItem(MissionAdapter.java:489)
	at us.shandian.giga.ui.adapter.MissionAdapter.access$200(MissionAdapter.java:68)
	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem.lambda$buildPopup$2(MissionAdapter.java:700)
	at us.shandian.giga.ui.adapter.-$$Lambda$MissionAdapter$ViewHolderItem$WMTXzvtjV6uhlyLXNG5QCw_ELrY.onMenuItemClick(lambda)
	at android.widget.PopupMenu.onMenuItemSelected(PopupMenu.java:142)
	at com.android.internal.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:735)
	at com.android.internal.view.menu.SubMenuBuilder.dispatchMenuItemSelected(SubMenuBuilder.java:81)
	at com.android.internal.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:149)
	at com.android.internal.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:874)
	at com.android.internal.view.menu.MenuPopupHelper.onItemClick(MenuPopupHelper.java:156)
	at android.widget.AdapterView.performItemClick(AdapterView.java:298)
	at android.widget.AbsListView.performItemClick(AbsListView.java:1128)
	at android.widget.AbsListView$PerformClick.run(AbsListView.java:2812)
	at android.widget.AbsListView.onTouchEvent(AbsListView.java:3583)
	at android.view.View.dispatchTouchEvent(View.java:7262)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2235)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1932)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2241)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1946)
	at android.widget.PopupWindow$PopupViewContainer.dispatchTouchEvent(PopupWindow.java:1592)
	at android.view.View.dispatchPointerEvent(View.java:7442)
	at android.view.ViewRootImpl.deliverPointerEvent(ViewRootImpl.java:3610)
	at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:3538)
	at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:4790)
	at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:4750)
	at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:4902)
	at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:179)
	at android.os.MessageQueue.nativePollOnce(Native Method)
	at android.os.MessageQueue.next(MessageQueue.java:125)
	at android.os.Looper.loop(Looper.java:140)
	at android.app.ActivityThread.main(ActivityThread.java:5297)
	at java.lang.reflect.Method.invokeNative(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:511)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:833)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:600)
	at dalvik.system.NativeStart.main(Native Method)

```
</p></details>
<hr>

**Comment by Brimont** [comment]
SHA-1
## Exception
* __User Action:__ ui error
* __Request:__ App crash, UI failure
* __Content Language:__ IT
* __Service:__ none
* __Version:__ 0.15.1


<details><summary><b>Crash log</b></summary><p>

```
java.lang.ClassCastException: org.schabi.newpipe.App cannot be cast to android.app.Activity
	at us.shandian.giga.ui.adapter.MissionAdapter$ChecksumTask.<init>(MissionAdapter.java:721)
	at us.shandian.giga.ui.adapter.MissionAdapter.handlePopupItem(MissionAdapter.java:489)
	at us.shandian.giga.ui.adapter.MissionAdapter.access$200(MissionAdapter.java:68)
	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem.lambda$buildPopup$2(MissionAdapter.java:700)
	at us.shandian.giga.ui.adapter.-$$Lambda$MissionAdapter$ViewHolderItem$WMTXzvtjV6uhlyLXNG5QCw_ELrY.onMenuItemClick(lambda)
	at android.widget.PopupMenu.onMenuItemSelected(PopupMenu.java:142)
	at com.android.internal.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:735)
	at com.android.internal.view.menu.SubMenuBuilder.dispatchMenuItemSelected(SubMenuBuilder.java:81)
	at com.android.internal.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:149)
	at com.android.internal.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:874)
	at com.android.internal.view.menu.MenuPopupHelper.onItemClick(MenuPopupHelper.java:156)
	at android.widget.AdapterView.performItemClick(AdapterView.java:298)
	at android.widget.AbsListView.performItemClick(AbsListView.java:1128)
	at android.widget.AbsListView$PerformClick.run(AbsListView.java:2812)
	at android.widget.AbsListView.onTouchEvent(AbsListView.java:3583)
	at android.view.View.dispatchTouchEvent(View.java:7262)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2235)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1932)
	at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2241)
	at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:1946)
	at android.widget.PopupWindow$PopupViewContainer.dispatchTouchEvent(PopupWindow.java:1592)
	at android.view.View.dispatchPointerEvent(View.java:7442)
	at android.view.ViewRootImpl.deliverPointerEvent(ViewRootImpl.java:3610)
	at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:3538)
	at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:4790)
	at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:4750)
	at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:4902)
	at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:179)
	at android.os.MessageQueue.nativePollOnce(Native Method)
	at android.os.MessageQueue.next(MessageQueue.java:125)
	at android.os.Looper.loop(Looper.java:140)
	at android.app.ActivityThread.main(ActivityThread.java:5297)
	at java.lang.reflect.Method.invokeNative(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:511)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:833)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:600)
	at dalvik.system.NativeStart.main(Native Method)

```
</p></details>
<hr>

**Comment by gkeegan** [comment]
Cannot confirm. What device is this on?

**Comment by theScrabi** [comment]
This information would be very helpfull.

**Comment by kapodamy** [comment]
try again with the lastest version v0.16

**Comment by gkeegan** [comment]
I think this should be closed. No info has been provided and it's for an older version of NewPipe.

**Comment by gkeegan** [comment]
This has been fixed, and will be refactored in kapodamy's #2149, so by all rights this should be closed.
